src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -530,6 +500,7 @@ private synchronized KeyPair getKeyPairFromAndroidKeyStore() throws GeneralSecur
     
     @TargetApi(18)
     private Object getKeyPairGeneratorSpec(X500Principal subj, Date start, Date end) {
+        // TODO: use real Spec.Builder instead of reflective methods
```

Comment:
Thanks @weijjia  for filing issue.
I don't really know the history why it uses reflective methods for KeyPairGeneratorSpec.Builder, I guess there were historical reasons behind (some problems on old versions or something else), and I think we need to know it before we do the change...